src/execution_plan/ops/op_distinct.c

```diff
@@ -150,7 +99,8 @@ static Record DistinctConsume(OpBase *opBase) {
 
```

Comment:
a comment outside the scope of this PR
I think we can move distinct operation(s) to the individual UNION streams and thus remove this awkward logic,
@jeffreylovitz what do you think ?

Reply:
We could create separate distinct operations for every UNION stream pretty easily - the first solution I can think of is annotating `RETURN` AST nodes with a marker denoting that they're a `UNION` subquery and thus should have a distinct op.

I don't think that this would be logically equivalent, however, as we would no longer be de-duping the outputs of all concatenated streams. Effectively, this change would convert a query like:
```
UNWIND [1, 1, 2] AS v RETURN v
UNION
UNWIND [1, 2] AS v RETURN v
```
into:
```
UNWIND [1, 1, 2] AS v RETURN DISTINCT v
UNION ALL
UNWIND [1, 2] AS v RETURN DISTINCT v
```
Which would produce 2 instances of `1` and 2 instances of `2`.